extensions/positron-python/python_files/posit/positron/positron_jedilsp.py

```diff
@@ -844,63 +948,32 @@ def positron_did_close_diagnostics(
-    # decorator causes the function to run in a separate thread thus a separate stack from pygls'
-    # exception handler.
-    try:
-        _publish_diagnostics(server, uri)
-    except Exception:
-        logger.exception(f"Failed to publish diagnostics for uri {uri}", exc_info=True)
-
-
-# Adapted from jedi_language_server/server.py::_publish_diagnostics.
-def _publish_diagnostics(server: PositronJediLanguageServer, uri: str) -> None:
-    """Helper function to publish diagnostics for a file."""
-    # The debounce decorator delays the execution by 1 second
-    # canceling notifications that happen in that interval.
-    # Since this function is executed after a delay, we need to check
-    # whether the document still exists
-    if uri not in server.workspace.text_documents:
-        return
-
-    doc = server.workspace.get_text_document(uri)
-
-    # Comment out magic/shell/help command lines so that they don't appear as syntax errors.
-    source = "\n".join(
-        (
-            f"#{line}"
-            if line.lstrip().startswith((_LINE_MAGIC_PREFIX, _SHELL_PREFIX, _HELP_PREFIX_OR_SUFFIX))
-            or line.rstrip().endswith(_HELP_PREFIX_OR_SUFFIX)
-            else line
-        )
-        for line in doc.lines
-    )
-
-    # Ignore all warnings during the compile, else they display in the console.
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore")
-        diagnostic = jedi_utils.lsp_python_diagnostic(uri, source)
-
-    diagnostics = [diagnostic] if diagnostic else []
-
-    server.publish_diagnostics(uri, diagnostics)
+@POSITRON.feature(NOTEBOOK_DOCUMENT_DID_SAVE)
+def positron_did_save_notebook_diagnostics(
+    server: PositronJediLanguageServer, params: DidSaveNotebookDocumentParams
+) -> None:
+    return did_save_notebook_diagnostics(server, params)
 
 
-def did_save_diagnostics(server: JediLanguageServer, params: DidSaveTextDocumentParams) -> None:
-    """Actions run on textDocument/didSave: diagnostics."""
-    _publish_diagnostics_debounced(server, params.text_document.uri)  # type: ignore - pyright bug
+@POSITRON.feature(NOTEBOOK_DOCUMENT_DID_CHANGE)
+def positron_did_change_notebook_diagnostics(
+    server: PositronJediLanguageServer, params: DidChangeNotebookDocumentParams
+) -> None:
+    return did_change_notebook_diagnostics(server, params)
 
 
-def did_change_diagnostics(server: JediLanguageServer, params: DidChangeTextDocumentParams) -> None:
-    """Actions run on textDocument/didChange: diagnostics."""
-    _publish_diagnostics_debounced(server, params.text_document.uri)  # type: ignore - pyright bug
+@POSITRON.feature(NOTEBOOK_DOCUMENT_DID_OPEN)
+def positron_did_open_notebook_diagnostics(
+    server: JediLanguageServer, params: DidOpenNotebookDocumentParams
+) -> None:
+    return did_open_notebook_diagnostics(server, params)
 
 
-def did_open_diagnostics(server: JediLanguageServer, params: DidOpenTextDocumentParams) -> None:
```

Comment:
do we not need these for .py scripts?

Reply:
Good question. We do still have the functions that used to call these, they now just call the jedi-language-server version directly instead of these.